lib/matplotlib/offsetbox.py

```diff
@@ -1028,6 +1042,14 @@ def __init__(self, loc,
         self.set_bbox_to_anchor(bbox_to_anchor, bbox_transform)
         self.set_child(child)
 
+        if is_string_like(loc):
+            try:
+                loc = self.codes[loc]
+            except KeyError:
+                raise RuntimeError('Unrecognized location "%s". Valid '
```

Comment:
Agreed, given RuntimeError is a catchall for anything not covered by other errors, and ValueError seems good here.